src/compile/mark-text.ts

```diff
@@ -72,18 +73,9 @@ export namespace text {
 
     // text
     if (model.has(TEXT)) {
-      if (model.fieldDef(TEXT).type === QUANTITATIVE) {
-        const format = model.config().mark.format;
-        // TODO: revise this line
-        const numberFormat = format !== undefined ? format : model.numberFormat(TEXT);
-
-        p.text = {
-          template: '{{' + model.field(TEXT, { datum: true }) +
-          ' | number:\'' + numberFormat + '\'}}'
-        };
-      } else {
-        p.text = { field: model.field(TEXT) };
-      }
+      p.text = { field: model.field(TEXT) };
+      const format = model.config().mark.format;
+      extend(p, model.formatMixins(TEXT, format));
```

Comment:
We generally avoid overriding and `p.text` is overridden for `Q` and `T`.

Better split this by `fieldDef.type`.   (And just call `formatMixins` for `Q` and `T`)